electrum-client/ElectrumClient.ts

```diff
@@ -605,4 +624,41 @@ export class ElectrumClient {
             tx.inputs.some(input => input.address && addresses.has(input.address))
             || tx.outputs.some(output => output.address && addresses.has(output.address)));
     }
+
+    private async callAgents<R>(call: AgentCall<R>, onAgentError?: AgentErrorCallback, stopAfterFirstResult?: true): Promise<R>;
+    private async callAgents<R>(call: AgentCall<R>, onAgentError: AgentErrorCallback | undefined, stopAfterFirstResult: false): Promise<R[]>;
+    private async callAgents<R>(call: AgentCall<R>, onAgentError?: AgentErrorCallback, stopAfterFirstResult: boolean = true): Promise<R | R[]> {
+        const untriedConnectedAgents: Set<Agent> = new Set(this.agents);
+        const results: R[] = [];
+        for (const agent of this.agents) {
+            try {
+                untriedConnectedAgents.delete(agent);
+                results.push(await call(agent));
+                if (stopAfterFirstResult) break;
+            } catch (error) {
+                if (onAgentError) {
+                    onAgentError(agent, error);
+                }
+
+                const { known: knownPeers, connected: connectedPeers } = this.getPeerStatistics();
+                if (untriedConnectedAgents.size || results.length || connectedPeers >= knownPeers) continue;
+
+                // All connected peers failed. Connect to a new one.
+                let newAgent: Agent | null = null;
+                let newAgentAttempts = 0;
+                while (!newAgent && newAgentAttempts < 3) {
+                    newAgentAttempts++;
+                    newAgent = await this.connectNewAgent();
+                }
+
+                if (!newAgent) break;
+
+                await this.waitForConsensusEstablished();
```

Comment:
This has no effect on the newly connected agent as long as the first agent is still in `synced` status.

This also plays into the whole consensus-between-multiple-agents topic that I described in the main review comment.